### PR TITLE
Enables Github publishing pipeline

### PR DESCRIPTION
**Pull request by nt3rp**
We want to be able to publish our own version of the SDK temporarily.

This PR removes the existing publishing pipeline (to avoid accidentally publishing to the maven main repository) and adds a pipeline to publish to Github.

New releases created in Github should create new packages.

### CODE REVIEW

File: .github/workflows/lint-test-sdk.yml [hunk 1]
```diff
@@ -14,7 +14,5 @@ jobs:
         with:
           java-version: '11'
           distribution: 'adopt'
-      - name: 'Set up GCP SDK'
-        uses: 'google-github-actions/setup-gcloud@v0'
```
Comment (nt3rp) [inline .github/workflows/lint-test-sdk.yml:18]: Comment (non-blocking): This wasn't strictly necessary to remove, but it appears to be unused and v0 is unsupported, so I removed it.

File: .github/workflows/publish-github.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      - uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Download test data
+        run: make test-data
+      - name: Publish package
+        run: mvn --batch-mode deploy
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (nt3rp) [inline .github/workflows/publish-github.yml:22]: Comment (non-blocking): Most of this comes from [Github's docs](https://docs.github.com/en/actions/publishing-packages/publishing-java-packages-with-maven#publishing-packages-to-github-packages), with some small modifications to replicate the existing publishing pipeline (e.g. Java version, running tests)

File: pom.xml [hunk 3]
```diff
@@ -77,8 +77,9 @@
 
     <distributionManagement>
         <snapshotRepository>
-            <id>ossrh</id>
-            <url>https://s01.oss.sonatype.org/content/repositories/snapshots</url>
+            <id>github</id>
+            <name>GitHub Packages</name>
+            <url>https://maven.pkg.github.com/GlossGenius/eppo-java-server-sdk</url>
```
Comment (nt3rp) [inline pom.xml:82]: Comment (non-blocking): This comes from [Github's docs](https://docs.github.com/en/actions/publishing-packages/publishing-java-packages-with-maven#publishing-packages-to-github-packages)

**Review by nicoraynaud — APPROVED** [review]
You could also temporarily set the workflow to be triggered from a PR if you want to test faster :)

File: .github/workflows/publish-github.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
```
Comment (dmontalvo-gg) [inline .github/workflows/publish-github.yml:16]: (non-blocking) Do we want to use Java 11? Most recent Java JDK LTS version is 21 (Sep 23), also Java 17 (Sep 21).

**Review by dmontalvo-gg — APPROVED** [review]
Looks good, only some questions about JDK version to build the Jar.

File: .github/workflows/publish-github.yml (hunk 4 above)
Comment (nt3rp) [reply .github/workflows/publish-github.yml:16]: It's the version that's defined in [the Maven configuration](https://github.com/GlossGenius/eppo-java-server-sdk/blob/main/pom.xml#L37), and was defined in the previous build pipeline 🤷🏻 

(If we make continued investments, then yes, we should likely update the Java version, but my current goal is to make the smallest changes required)

**Pull request merged**